libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -199,3 +199,39 @@ def _insert_embeddings(
         except DBAPIError as e:
             logging.error(e.__cause__)
         return ids
+
+    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> Optional[bool]:
+        """Delete embeddings in the vectorstore by the ids.
+
+        Args:
+            ids: List of IDs to delete.
+            kwargs: vectorstore specific parameters.
+
+        Returns:
+            Optional[bool]
+        """
+
+        if ids is None or len(ids) == 0:
+            logging.info("Empty list of ids provided")
+            raise ValueError("No ids provided to delete.")
```

Comment:
If ids is required, why does the signature say its Optional?

Reply:
Yeah, according to the signature ids are optional.

I have raised an error as a precheck, but you are right. I will instead return False and log it as an log.info.

Reply:
Can you please make the change to return False in that case? With respect to the kind of log, log.debug might be useful for null or empty lists and log.info for ids not found. You are right, returning error would give wrong indication but we need to better inform the user

Reply:
Yeah, will change do the changes